### DISCUSSION TITLE
Idea: Play/Stop button for continuous grow

### DISCUSSION THREAD

**Discussion by marianbasti** [inca/blender-differential-growth#1] [Ideas]
And avoiding carpal tunnel syndrome 👍 
Parameters for limiting the process could be number of steps, edge radius and size/distance to object

**Comment by mty39** [comment]
Definitely! And possibility to animate parameters, please .)
When each step push time one frame forward.

**Reply by fewerjunk** [reply]
Perhaps a good idea to link animation to the Scene time node: Geometry nodes (or perhaps something comparable using Animation nodes):

https://www.youtube.com/watch?v=X41wo3ExlJM

**Comment by ewerybody** [comment]
After playing with it for a while I'd think this would be a **terrible** idea! At least for how it's currently implemented.
Depending on the settings the groth already might either stop entirely, explode completely or take forever for the next step..

@fewerjunk implementing the whole thing upon geometry nodes simulations migth be worth looking into tho! But that'd be a different addon?

**Reply by inca** [reply]
Yep, absolutely agree. I tried a number of approaches, but so far none works, because the algorithm is just too computationally heavy and non-deterministic (i.e. slightest change of any of the parameters results in drastically different result).

Simulation nodes look very promising: they seem to include caches and are generally faster than Python APIs. But as I mentioned elsewhere, lack of selective edge subdivision in nodes is currently a major blocker. If anyone knows any good resource on how to subdivide edges/faces in nodes (even if low-level), please send them my way 🙏

**Reply by ewerybody** [reply]
Hey @inca! I was thinking about this a little while, tinkered with an idea to explore the settings space:
* Maybe one can have a grid created with copies of the source mesh and in x and y directions you'll have variations of 2 or more settings
* there could be some interface where one can draw curves for the different settings to have them changed over the steps
* have some `max calc time` that causes the iteration to break when reached

(well, these first 2 could probably be an addon for the addon ;))

After all: Maybe it's not too complicated to implement this as a c/c++ plugin that can be talked to via python?
Btw: have you profiled it yet? What's the heaviest bottleneck?
